62.CAD/vertex_shader.hlsl

```diff
@@ -2,18 +2,213 @@
+    else t = num / den;
+    return t;
+}
+
+float2 QuadraticBezier(float2 p0, float2 p1, float2 p2, float t)
+{
+    float tt = t * t;
+    float oneMinusT = 1.0 - t;
+    float oneMinusTT = oneMinusT * oneMinusT;
+    float2 position = p0 * oneMinusTT + 2.0 * p1 * oneMinusT * t + p2 * tt;
+    return position;
+}
+
+
+float2 CalculateIntersection(float2 line1Start, float2 line2Start, float2 line1End, float2 line2End)
+{
+    float2 intersectionPoint;
+
+    float2 line1Direction = normalize(line1End - line1Start);
+    float2 line2Direction = normalize(line2End - line2Start);
+
+    float line1Denominator = line2Direction.y * line1Direction.x - line2Direction.x * line1Direction.y;
+
+    if (abs(line1Denominator) < 0.0001)
+    {
+        // Lines are parallel, return a default value or handle accordingly.
+        intersectionPoint = float2(0, 0); // Default value, adjust as needed.
+    }
+    else
+    {
+        float2 line2StartToLine1Start = line1Start - line2Start;
+        float line2Numerator = line2Direction.x * line2StartToLine1Start.y - line2Direction.y * line2StartToLine1Start.x;
+
+        float t = line2Numerator / line1Denominator;
+        intersectionPoint = line1Start + t * line1Direction;
+    }
+
+    return intersectionPoint;
+}
+
+float3 normalize(float3 A) {
+    return A / sqrt(dot(A, A));
+}
+
+
+bool estimateTransformation(float2 p01, float2 p11, float2 p21, out float2 translation, out float2x2 rotation)
+{
+
+
+    float2 p1 = p11 - p01;
+    float2 p2 = p21 - p01;
+
+
+
+    float2 a = p2 - 2.0 * p1;
+    float2 b = 2.0 * p1;
+
+
+
+    float2 mean = a / 3.0 + b / 2.0;
+
+
+
+    float axy = a.x * a.y;
+    float bxy = a.x * b.y + b.x * a.y;
+    float cxy = b.x * b.y;
+
+
+
+    float2 aB = a * a;
+    float2 bB = a * b * 2.0;
+    float2 cB = b * b;
+
+
+    float xy = axy / 5.0 + bxy / 4.0 + cxy / 3.0;
+    float xx = aB.x / 5.0 + bB.x / 4.0 + cB.x / 3.0;
+    float yy = aB.y / 5.0 + bB.y / 4.0 + cB.y / 3.0;
+
+    float cov_00 = xx - mean.x * mean.x;
+    float cov_01 = xy - mean.x * mean.y;
+    float cov_11 = yy - mean.y * mean.y;
+
+
+    float eigen_a = 1.0;
+    float eigen_b_neghalf = -(cov_00 + cov_11) * -0.5;
+    float eigen_c = (cov_00 * cov_11 - cov_01 * cov_01);
+
+    float discr = eigen_b_neghalf * eigen_b_neghalf - eigen_a * eigen_c;
+    if (discr <= 0.0)
+        return false;
+
+    discr = sqrt(discr);
+
+    float lambda0 = (eigen_b_neghalf - discr) / eigen_a;
+    float lambda1 = (eigen_b_neghalf + discr) / eigen_a;
+
+
+    float2 eigenvector0 = float2(cov_01, lambda0 - cov_00);
+    float2 eigenvector1 = float2(cov_01, lambda1 - cov_00);
+
+    // result
+
+    rotation[0] = normalize(eigenvector0);
+    rotation[1] = normalize(eigenvector1);
+
+    translation = mean + p01;
+
+    return true;
+}
+float bezierEval(float v0, float v1, float v2, float t)
+{
+    float s = 1.0 - t;
+
+    return v0 * (s * s) +
+        v1 * (s * t * 2.0) +
+        v2 * (t * t);
+}
```

Comment:
Did you just reimplement `QuadraticBezier` above in a different way?

Reply:
it computes it in one dimension as the min and max's for the OBB have different T values